packages/wallet/src/rollup-client.ts

```diff
@@ -46,51 +57,77 @@ export class RollupClient {
   }
 
   /**
-   * Connects to an aggregator using a provided rpcClient
-   * @param rpcClient the rpcClient to use -- normally it's a SimpleClient
+   * Gets the state for the provided account.
+   * @param account the account whose state will be retrieved.
+   * @returns the SignedStateReceipt for the account
    */
   public async getState(account: Address): Promise<SignedStateReceipt> {
-    return this.rpcClient.handle<SignedStateReceipt>(
-      AGGREGATOR_API.getState,
-      account
-    )
+    const receipt: SignedStateReceipt = await this.rpcClient.handle<
+      SignedStateReceipt
+    >(AGGREGATOR_API.getState, account)
+    this.verifyTransactionReceipt(receipt)
+    return receipt
   }
 
   public async getUniswapBalances(): Promise<SignedStateReceipt> {
     return this.getState(UNISWAP_ADDRESS)
   }
 
   public async sendTransaction(
-    transaction: Transaction,
+    transaction: RollupTransaction,
     account: Address
-  ): Promise<SignedTransactionReceipt> {
+  ): Promise<SignedStateReceipt[]> {
     const signature = await this.signatureProvider.sign(
       account,
-      serializeObject(transaction)
-    )
-    return this.rpcClient.handle<SignedTransactionReceipt>(
-      AGGREGATOR_API.applyTransaction,
-      {
-        signature,
-        transaction,
-      }
+      abiEncodeTransaction(transaction)
     )
+    const receipts: SignedStateReceipt[] = await this.rpcClient.handle<
+      SignedStateReceipt[]
+    >(AGGREGATOR_API.applyTransaction, {
+      signature,
+      transaction,
+    })
+
+    return receipts
   }
 
   public async requestFaucetFunds(
-    transaction: Transaction,
+    transaction: RollupTransaction,
     account: Address
-  ): Promise<SignedTransactionReceipt> {
+  ): Promise<SignedStateReceipt> {
     const signature = await this.signatureProvider.sign(
       account,
       serializeObject(transaction)
     )
-    return this.rpcClient.handle<SignedTransactionReceipt>(
-      AGGREGATOR_API.requestFaucetFunds,
-      {
-        signature,
-        transaction,
-      }
+    const receipt: SignedStateReceipt = await this.rpcClient.handle<
+      SignedStateReceipt
+    >(AGGREGATOR_API.requestFaucetFunds, {
+      signature,
+      transaction,
+    })
+    this.verifyTransactionReceipt(receipt)
+    return receipt
+  }
+
+  private verifyTransactionReceipt(receipt: SignedStateReceipt): void {
+    if (
+      receipt.signature === EMPTY_AGGREGATOR_SIGNATURE &&
+      receipt.stateReceipt.leafID === NON_EXISTENT_LEAF_ID
+    ) {
+      return
+    }
+
+    const signer = this.signatureVerifier.verifyMessage(
+      abiEncodeStateReceipt(receipt.stateReceipt),
+      receipt.signature
     )
+    if (signer !== AGGREGATOR_ADDRESS) {
+      log.error(
+        `Received invalid SignedStateReceipt from the Aggregator: ${serializeObject(
+          receipt
+        )}`
+      )
+      throw new SignatureError()
+    }
```

Comment:
Ah didn't realize you'd already done this during our conversation! This is great and then we can move the signature verification into the OVM